testsuite/ui/widgets/buttons.py

```diff
@@ -25,9 +25,14 @@ class ThreescaleButton(widgetastic_patternfly4.Button):
 
     def __init__(self, parent, text, classes, **kwargs):
         base = "(self::a or self::button or (self::input and (@type='button' or @type='submit')))"
-        classes = " and ".join(f"contains(@class, {quote(i)})" for i in classes)
-        conditions = f"(({classes}) or contains(text(), {quote(text)}))"
-        super().__init__(parent, locator=f".//*[{base} and {conditions}]", **kwargs)
+        class_conditions = " and ".join(f"contains(@class, {quote(cls)})" for cls in classes)
+
+        conditions = f"({class_conditions} or contains(text(), {quote(text)}))"
+        if "id" in kwargs:
```

Comment:
why not specify id as a optional parameter?

Reply:
@pehala this particular line is result of previous review, should be clear from that